agent/src/main/java/com/teamscale/jacoco/agent/convert/Converter.java

```diff
@@ -64,7 +65,9 @@ public void runJaCoCoReportGeneration() throws IOException {
 				wrap(logger));
 
 		try (Benchmark benchmark = new Benchmark("Generating the XML report")) {
-			generator.convert(new Dump(sessionInfo, executionDataStore), Paths.get(arguments.outputFile));
+			generator.convert(new Dump(sessionInfo, executionDataStore), Paths.get(arguments.outputFile), true);
+		} catch (EmptyReportException e) {
+			logger.error("EmptyReportException thrown, this should never happen while using the converter tool.");
```

Comment:
Can the coverage file only be empty, if we specify wrong includes and excludes? Or is it also possible that the exec file already does not contain any coverage? If the second is the case, I'd prefer the solution as is, as I'd expect as user that an "empty" exec file generates an "empty" xml file.
The difference that I see between convert and our agent is, that with the agent we are controlling the entire process (converting **and** collecting) whereas with convert we only control converting.

Reply:
While it's true that one could expect an empty report for an empty exec file, I think an empty exec file is never the result the user wants (after all: why record coverage if you already know you'll get an empty report?) but always a configuration mistake by the user. I have seen it multiple times that users misconfigure JaCoCo, get an empty report, upload it to Teamscale and then say "Teamscale is broken, it doesn't show my coverage". So I'd much rather warn them right here that what they're doing is broken on their end and how to fix it rather than having to deal with it in a support request.

Reply:
Ok, I suppose our agent will only be used in conjunction with Teamscale anyways, so I think I agree now. Will remove the boolean flag and change the log message here.